test/unit/ngsiv2/configurationsMqtt-test.js

```diff
@@ -0,0 +1,228 @@
+                setTimeout(function () {
+                    should.exist(configurationReceived);
+                    done();
+                }, 100);
+            });
+        });
+    });
+
+    describe('When a subscription request is received in the IoT Agent', function () {
+        const values = 'subscription|pollingInterval|publishInterval';
+        let configurationReceived;
+
+        beforeEach(function () {
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartgondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post(
+                    '/v2/subscriptions',
+                    utils.readExampleFile('./test/unit/ngsiv2/subscriptionRequests/configurationsMqttRequest.json')
+                )
+                .reply(201, null, { Location: '/v2/subscriptions/51c0ac9ed714fb3b37d7d5a8' });
+
+            mqttClient.subscribe('/1234/MQTT_device_1/configuration/values', null);
+
+            configurationReceived = false;
+        });
+
+        afterEach(function (done) {
+            mqttClient.unsubscribe('/1234/MQTT_device_1/configuration/values', null);
+
+            done();
+        });
+
+        it('should create a subscription in the ContextBroker', function (done) {
+            mqttClient.publish('/1234/MQTT_device_1/configuration/commands', values, null, function (error) {
+                setTimeout(function () {
+                    contextBrokerMock.done();
+                    done();
+                }, 100);
+            });
+        });
+
+        it('should update the values in the MQTT topic when a notification is received', function (done) {
+            const optionsNotify = {
+                url: 'http://localhost:' + config.iota.server.port + '/notify',
+                method: 'POST',
+                json: utils.readExampleFile(
+                    './test/unit/ngsiv2/subscriptionRequests/configurationsMqttNotification.json'
+                ),
+                headers: {
+                    'fiware-service': 'smartgondor',
+                    'fiware-servicepath': '/gardens'
+                }
+            };
+
+            mqttClient.on('message', function (topic, data) {
+                const result = utils.parseConfigurationResponse(data.toString());
+
+                configurationReceived = result.pollingInterval === '60' && result.publishInterval === '600';
+            });
+
+            mqttClient.publish('/1234/MQTT_device_1/configuration/commands', values, null, function (error) {
+                setTimeout(function () {
+                    request(optionsNotify, function (error, response, body) {
+                        setTimeout(function () {
+                            configurationReceived.should.equal(true);
+                            done();
+                        }, 100);
+                    });
+                }, 100);
+            });
```

Comment:
This part does not work. The expeted behavoiurf is the following:
1. The test subscribe to `/1234/MQTT_device_1/configuration/values` and set up the enviroment - OK
2. The test publish to `/1234/MQTT_device_1/configuration/commands` - OK
3. The test POST a notification payload to IoTA's /notify endpoint - OK
4. The IoTA should publish to  '/1234/MQTT_device_1/configuration/values' the information - NOK

In the 4th step, the information published in the topic does not seem the right one. Having the following notification mocked:

```json
{
  "subscriptionId": "51c0ac9ed714fb3b37d7d5a8",
  "data": [
    {
      "id": "MQTT Device 1",
      "type": "AnMQTTDevice",
      "pollingInterval": {
        "type": "string",
        "value": "60",
        "metadata": {
        }
      },
      "publishInterval": {
        "type": "number",
        "value": "600",
        "metadata": {
        }
      }
    }
  ]
}
``` 

The IoTA is publishing:

```
MQTT_device_1@configuration|0=60|1=606|2=600|dt=20210618T173202Z
```

It seems that this part of the code puts directly the index of the element (0) instead of the name of the attribute (pollingInterval)

Reply:
Could it be a bug in the configurations functionality? How it works in the parallel NGSIv1 test?

Reply:
Checking V1 test version, the result seems the expected, with the name of the attributes

```
MQTT_device_1@configuration|pollingInterval=60|publishInterval=600|dt=20210621T131150Z
```

Reply:
Fixed in PR #496. Last commit in this PR changes xit() to it(). NTC